lib/tapioca/executor.rb

```diff
@@ -25,24 +28,62 @@ def initialize(queue, number_of_workers: nil)
       @items_per_worker = T.let((queue.length.to_f / @number_of_workers).ceil, Integer)
     end
 
-    sig { params(block: T.proc.params(item: T.untyped).void).void }
+    sig do
+      type_parameters(:T).params(
+        block: T.proc.params(item: T.untyped).returns(T.type_parameter(:T))
+      ).returns(T.nilable(T::Array[T.type_parameter(:T)]))
+    end
     def run_in_parallel(&block)
       # If we only have one worker selected, it's not worth forking, just run sequentially
-      if @number_of_workers == 1
-        block.call(@queue.shift) until @queue.empty?
-        return
-      end
+      return @queue.map { |item| block.call(item) }.compact if @number_of_workers == 1
+
+      # Create an IO pipe to communicate the return value of the parallelized block back from the workers to the main
+      # process. It's important to only create pipes if we're running with more than one worker, otherwise the tests
+      # fail with a "too many open files" error.
```

Comment:
> otherwise the tests fail with a "too many open files" error

This is concerning. How do we reach the `$ ulimit -n` with only one worker? Are we missing a `close` somewhere? Or could this be related to the `IsolationHelper` we use?